File: 637-average-of-levels-in-binary-tree/average-of-levels-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var averageOfLevels = function(root) {
    const averages=[];
    const queue=[root];
    while (queue.length>0){
        let levelSize=queue.length;
        let levelSum=0;
        for(let i=0; i<levelSize; i++){
            const node=queue.shift();
            levelSum+=node.val;

            if(node.left) queue.push(node.left);
            if(node.right) queue.push(node.right);
        }averages.push(levelSum /levelSize);
    }return averages;
};

File: 637-average-of-levels-in-binary-tree/average-of-levels-in-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<double> AverageOfLevels(TreeNode root) {
        IList<double> result = new List<double>();
        if (root == null) return result;

        Queue<TreeNode> queue = new Queue<TreeNode>();
        queue.Enqueue(root);

        while (queue.Count > 0) {
            int levelSize = queue.Count;
            double sum = 0;

            for (int i = 0; i < levelSize; i++) {
                TreeNode currentNode = queue.Dequeue();
                sum += currentNode.val;

                if (currentNode.left != null) queue.Enqueue(currentNode.left);
                if (currentNode.right != null) queue.Enqueue(currentNode.right);
            }

            result.Add(sum / levelSize);
        }

        return result;
    }
}